diff -urN klog-2.4.2/CMakeLists.txt klog-2.4.2-cmakelist-fixes/CMakeLists.txt
--- klog-2.4.2/CMakeLists.txt	2025-08-14 12:01:43.000000000 +0100
+++ klog-2.4.2-cmakelist-fixes/CMakeLists.txt	2025-11-19 15:04:09.157468283 +0000
@@ -1,12 +1,18 @@
-cmake_minimum_required(VERSION 3.16)
-project(KLog LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.5)
+project(KLog VERSION 2.4.2 LANGUAGES CXX)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/src//version.h)
 
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
 set(CMAKE_CXX_STANDARD 17)
+set(APP_VERSION "${PROJECT_VERSION}")
+set(APP_PKGVERSION "2.4.2-RC3")
 
-find_package(Qt5 COMPONENTS Widgets Sql Xml Network REQUIRED)
+find_package(Qt6 COMPONENTS Charts Core Gui Qml Widgets Sql Location Network Positioning PrintSupport QuickWidgets SerialPort Xml REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(HAMLIB REQUIRED hamlib)
+include_directories(${HAMLIB_INCLUDE_DIRS})
 
 # Gather all sources, headers, and UI files
 set(SOURCES
@@ -33,7 +39,7 @@
     src/inputwidgets/mainwindowinputothers.cpp
     src/inputwidgets/mainwindowinputeqsl.cpp
     src/inputwidgets/mainwindowinputqsl.cpp
-    src/inputwidgets/mainwindowsattab.cpp\
+    src/inputwidgets/mainwindowsattab.cpp
     src/searchmodel.cpp
     src/searchwindow.cpp
     src/setupdialog.cpp
@@ -54,7 +60,7 @@
     src/fileawardmanager.cpp
     src/database.cpp
     src/dataproxy_sqlite.cpp
-    src/downloadcty.cpp    
+    src/downloadcty.cpp
     src/locator.cpp
     src/qso.cpp
     src/awards.cpp
@@ -151,7 +157,7 @@
 	src/inputwidgets/mainwindowinputeqsl.h
 	src/inputwidgets/mainwindowinputqsl.h
 	src/inputwidgets/mainwindowsattab.h
-    klogdefinitions.h
+	src/klogdefinitions.h
 	src/lotwutilities.h
 	src/mainqsoentrywidget.h
 	src/mainwindow.h
@@ -202,18 +208,21 @@
 	src/widgets/showadifimportwidget.h
     src/updatesettings.h
     src/world.h
+    src/version.h
     # ... add other .h files
 )
 
 set(UIS
     # ... add .ui files here, if any
-	src/klog.ui
+	#src/klog.ui
 )
 
 set(RCCS
     # ... add .qrc files here, if any
+    src/klog.qrc
 )
 
+
 add_executable(klog
     ${SOURCES}
     ${HEADERS}
@@ -221,15 +230,28 @@
     ${RCCS}
 )
 
+target_include_directories(klog PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+find_package(Hamlib QUIET)
+if(Hamlib_FOUND)
+    target_link_libraries(klog Hamlib::Hamlib)
+else()
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(HAMLIB REQUIRED hamlib)
+    include_directories(${HAMLIB_INCLUDE_DIRS})
+    target_link_libraries(klog ${HAMLIB_LIBRARIES})
+endif()
+
 target_link_libraries(klog
 	Qt6::Core
-    Qt6::Gui
-    Qt6::Sql
-    Qt6::Network
-    Qt6::Serialport
-    Qt6::Printsupport
-    Qt6::Charts
-    Qt6::Quickwidgets
-    Qt6::positioning
-    
-)
\ No newline at end of file
+	Qt6::Gui
+	Qt6::Sql
+	Qt6::Network
+	Qt6::SerialPort
+	Qt6::PrintSupport
+	Qt6::Charts
+	Qt6::QuickWidgets
+	Qt6::Positioning
+	Qt6::Xml
+	${HAMLIB_LIBRARIES}
+)
diff -urN klog-2.4.2/src/main.cpp klog-2.4.2-cmakelist-fixes/src/main.cpp
--- klog-2.4.2/src/main.cpp	2025-08-14 12:01:43.000000000 +0100
+++ klog-2.4.2-cmakelist-fixes/src/main.cpp	2025-11-19 14:36:15.130393100 +0000
@@ -37,6 +37,9 @@
 #include "startwizard.h"
 #include "mainwindow.h"
 #include "utilities.h"
+//#ifdef KLOG_USE_VERSION_H
+#include "version.h"
+//#endif
 
 
 int showNoDB()
